cmd/istioctl/istioctl_test.go

```diff
@@ -15,159 +15,359 @@
+		},
+		{
+			name:      "TestUpdateErrorsWithArg",
+			command:   "put",
+			file:      "testdata/two-route-rules.yaml",
+			wantError: "replace takes no arguments",
+			arg:       []string{"arg-im-a-pirate"},
+		},
+		{
+			name:      "TestUpdateNoFile",
+			command:   "put",
+			wantError: "nothing to replace",
+		},
+		{
+			name:              "TestDeleteSuccessWithFile",
+			command:           "delete",
+			deleteKeySliceReq: true,
+			file:              "testdata/two-route-rules.yaml",
+		},
+		{
+			name:      "TestDeleteArgsErrorWithFile",
+			command:   "delete",
+			file:      "testdata/two-route-rules.yaml",
+			arg:       []string{"arg-im-a-pirate"},
+			wantError: "delete takes no arguments when the file option is used",
+		},
+		{
+			name:      "TestDeleteNoArgsErrorWithoutFile",
+			command:   "delete",
+			wantError: "provide configuration type and name or -f option",
+		},
+		{
+			name:      "TestDeleteErrorsPassedBackWithFile",
+			command:   "delete",
+			file:      "testdata/two-route-rules.yaml",
+			wantError: "an error",
+		},
+		{
+			name:              "TestDeleteSuccessWithoutFile",
+			command:           "delete",
+			arg:               []string{"route-rule", "test-v1"},
+			deleteKeySliceReq: true,
+		},
+		{
+			name:      "TestDeleteErrorsPassedBackWithoutFile",
+			command:   "delete",
+			arg:       []string{"route-rule", "test-v1"},
+			wantError: "an error",
+		},
+		{
+			name:      "TestGetNoArgs",
+			command:   "get",
+			wantError: "specify the type of resource to get. Types are destination-policy, ingress-rule, route-rule",
+		},
+		{
+			name:      "TestListPassesBackErrors",
+			command:   "get",
+			arg:       []string{"route-rule"},
+			wantError: "an error",
+			outFormat: "short",
+		},
+		{
+			name:            "TestGetRouteRule",
+			command:         "get",
+			arg:             []string{"route-rule"},
+			configKeyMapReq: true,
+			outFormat:       "short",
+		},
+		{
+			name:            "TestGetRouteRules",
+			command:         "get",
+			arg:             []string{"route-rules"},
+			configKeyMapReq: true,
+			outFormat:       "short",
+		},
+		{
+			name:            "TestGetDestPolicy",
+			command:         "get",
+			arg:             []string{"destination-policy"},
+			configKeyMapReq: true,
+			outFormat:       "short",
+		},
+		{
+			name:            "TestGetDestPolicies",
+			command:         "get",
+			arg:             []string{"destination-policies"},
+			configKeyMapReq: true,
+			outFormat:       "short",
+		},
+		{
+			name:            "TestGetYAML",
+			command:         "get",
+			arg:             []string{"route-rule"},
+			configKeyMapReq: true,
+			outFormat:       "yaml",
+		},
+		{
+			name:            "TestGetNotYAMLOrShort",
+			command:         "get",
+			arg:             []string{"route-rule"},
+			configKeyMapReq: true,
+			outFormat:       "not-an-output-format",
+			wantError:       "unknown output format not-an-output-format. Types are yaml|short",
```

Comment:
Comparing specific error message strings is brittle. Better to define a new error type if its important to check for a specific returned error. Or make `wantError` a `bool` and remove lines 369-371 below.